src/NuGet.Clients/PackageManagement.VisualStudio/Utility/EnvDTEProjectUtility.cs

```diff
@@ -1204,6 +1202,43 @@ private static string GetPattern(string token)
             return token == "*" ? @"(.*)" : @"(" + token + ")";
         }
 
+        /// <summary>
+        /// A DTE specific helper method that validates a path to ensure that it 
+        /// could be for a file as opposed to a URL or other invalid path, and
+        /// not for a directory. This is used to help determine if a value returned
+        /// from DTE is a directory or file, since the file may still be in 
+        /// memory and not yet written to disk File.Exists will not work.
+        /// </summary>
+        private static string GetPotentialFullPathOrNull(string path)
+        {
+            string fullPath = null;
+
+            try
+            {
+                if (!string.IsNullOrEmpty(path))
+                {
+                    // Attempt to get the full path
+                    fullPath = Path.GetFullPath(path);
+
+                    if (Directory.Exists(fullPath))
```

Comment:
perhaps add a comment here that lists the cases that we do know that will return a directory (website, js winrt, ???)